### ISSUE TITLE
1.9 features, items, blocks, generation etc.

### ISSUE THREAD

**Issue by mmdk95** [cuberite/cuberite#2406]
I know 1.8 isn't even fully implemented yet, but I am just placing it right here:
http://minecraft.gamepedia.com/15w31a

**Comment by SphinxC0re** [comment]
Not even all the biomes from 1.7 are implemented :dash:

**Comment by mmdk95** [comment]
@SphinxC0re oh didn't even know that Cx

**Comment by SamJBarney** [comment]
Which ones are not implemented?

**Comment by SphinxC0re** [comment]
https://github.com/cuberite/cuberite/milestones/1.7%20biomes

**Comment by jammet** [comment]
Reading that list of biomes gets me quite dreamy. I love some of those. Hope to see them in Cuberite.

**Comment by SphinxC0re** [comment]
@SamJBarney and me were having a discussion about how to rewrite the biome code. We talked about putting ever biome definition into it's own class. After that, it should be a piece of cake :smile:

**Comment by madmaxoft** [comment]
What would you put in such a class? And how would you then make it work with our composable generator?

**Comment by SphinxC0re** [comment]
You would throw a cBiomeSet into the generator so you can define which biomes are generated or not

**Comment by SphinxC0re** [comment]
Such a class would follow up the definition of a biome:
*Biomes are regions or the terrains in a Minecraft world with varying geographical features, flora, heights, temperatures, sky colours and foliage colours. Introduced in the Halloween Update, biomes separate every generated world into different environments, paralleling the real world; examples of biomes include the forest, jungle, desert and tundra. Prior to the Halloween Update, every world had only a single theme, either grassy or
Minecraft-jungle-shot

A Jungle Biome.
snowy.

The term biome is analogous to its scientific usage: on Earth, a biome is climatically and geographically defined by distinctive communities of plants, animals and soil organisms supported by similar climatic conditions. They are often referred to as ecosystems.*
but it wouldn't contain the positional definitions

**Comment by madmaxoft** [comment]
Still, it says nothing technical - what will the new class take care of? Will it define individual trees? Will it have village prefabs? Will it provide terrain composition? And what if the admin sets the world generator differently? There are generators that are not biome-aware, or fill biomes with other blocks than one would expect (DebugBiomes).

**Comment by SamJBarney** [comment]
The biome class would store the information about the biome, such as temperature and humidity, along with what trees and plants have been registered to grow naturally in that biome.

The idea is still pretty new, and we're still figuring out how to best implement it. We're open to suggestions.

**Comment by SphinxC0re** [comment]
Thank you @SamJBarney I totally forgot that really good answer you already gave *me* :clap:

**Comment by madmaxoft** [comment]
Temperature and humidity are not used by the server at all. What trees - this depends on the generator, if the trees are allowed at all, and how about custom trees provided in a prefab form? Same with plants.

As far as I see, there's not much use for specific biome classes, mostly because the generator's customizability breaks most of the intended uses.

**Comment by alwaysontop617** [comment]
Why Everytime I try and connect I get this warning:

`[18:24:06] Client "::ffff:192.168.1.154" uses an unsupported protocol (lengthed, version 107 (0x6b))`

I am using Minecraft 1.9.

**Comment by tigerw** [comment]
Indeed, protocol 1.9 support is entirely non-existent. i cri evrytiem

**Comment by alwaysontop617** [comment]
@tigerw  : Sorry to hear that

**Comment by jammet** [comment]
We're all suffering.

**Comment by mathwhiz1212** [comment]
Any word on how long until items are functional? Shields block damage. etc.
